dev-libs/rccl/rccl-5.7.1.ebuild

```diff
@@ -17,7 +17,7 @@ SLOT="0/$(ver_cut 1-2)"
 IUSE="test"
 
 RDEPEND="
-	dev-util/hip
+	dev-util/hip:${SLOT}
```

Comment:
Using sub slot to lock hip version here might be too strict, since the SLOT of hip-5.x.y is 5.x, meaning it restricts to a minor version. That is used to lock rocm-device-libs/rocm-comgr/hip, and rocBLAS/rocSOLVER, etc. From experiences it seems `sci-libs/*` and hip is compatible in the same major version.

Ref: https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=dde5dd2ff1162e22474fc87ccf6098943fee0caa

Reply:
I personally suggests using "=dev-util/hip-5*" here.

Reply:
Ok, `=dev-util/hip-5*` seems to work fine.

Reply:
@littlewu2508 I had the same issue as you described here https://github.com/ROCm/ROCm-CompilerSupport/issues/53#issuecomment-1807354795 but in runtime not compile time with opencl. The linking error was also evident in output from clinfo command. unmerging and recompiling hip, rocm-comgr, rocm-device-libs, rocBLAS in order fixed the issue

Reply:
> hip, rocm-comgr, rocm-device-libs, rocBLAS

Thanks for the report, while the order confused me.

The issue is on opencl side, it should have nothing to do with hip and rocBLAS. So why the emerge order of hip and rocBLAS matters, and why hip should be in first place? hip should depend on rocm-device-libs and rocm-comgr.

Also, rocm-comgr should depend on rocm-device-libs. How did you manage to make a clean installation in rocm-comgr -> rocm-device-libs order?

Basically, rocm-comgr and rocm-device-libs is "part of llvm" but not upstreamed for some reasons, so they strongly correlated to llvm major versions, and their own version should also be consistent, and their order matters -- it's like you need first install/upgrade llvm and then clang.

Reply:
@littlewu2508 oh sorry I just realised now that my wording was misleading. the order was the unmerging order. However shortly after my report I noticed the same issue showing up again and I couldn't explain why. rocminfo again has the NULL errors.

I first installed rocm-device-libs and then rocm-comgr then lastly hip